lib/node_modules/@stdlib/stats/base/dists/planck/pmf/lib/factory.js

```diff
@@ -0,0 +1,77 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isNonNegativeInteger = require( '@stdlib/math/base/assert/is-nonnegative-integer' );
+var constantFunction = require( '@stdlib/utils/constant-function' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var exp = require( '@stdlib/math/base/special/exp' );
+var expm1 = require( '@stdlib/math/base/special/expm1' );
+
+
+// MAIN //
+
+/**
+* Returns a function for evaluating the probability mass function (PMF) for a Planck distribution with shape parameter `lambda`.
+*
+* @param {number} lambda - shape parameter
```

Comment:
```suggestion
* @param {NonNegativeNumber} lambda - shape parameter
```

Yes, we handle `lambda < 0.0`, but the intent is that users should provide a nonnegative number.